src/Microsoft.AspNet.Razor.Runtime/TagHelpers/TagHelperContext.cs

```diff
@@ -37,6 +42,15 @@ public class TagHelperContext
         public IDictionary<string, object> AllAttributes { get; }
 
         /// <summary>
+        /// Gets the collection of items used to communicate with child <see cref="ITagHelper"/>s.
+        /// </summary>
+        /// <remarks>
+        /// This <see cref="IDictionary{object, object}"/> is copy-on-write in order to not affect parent
+        /// <see cref="ITagHelper"/>s.
```

Comment:
misleading since communication back to parent `ITagHelper`s is completely supported. suggest `in order to ensure items added to this collection are visible only to other <see cref="ITagHelper"/>s targeting this or child elements.`

might not need "this or" since, while it'll work (modulo order), it's not a primary use case.